tests/models/test_abstract.py

```diff
@@ -161,25 +178,80 @@ def test_time_step(bmi, model: MockedModel):
 
 
 def test_output_var_names(bmi, model: MockedModel):
-    bmi.get_output_var_names.return_value = ('discharge', )
+    bmi.get_output_var_names.return_value = ('discharge',)
 
     names = model.output_var_names
 
-    assert names == ('discharge', )
+    assert names == ('discharge',)
 
 
 def test_get_value_as_xarray(model: MockedModel):
     expected = xr.DataArray(
-            data=[[1.0, 2.0], [3.0, 4.0]],
-            coords={
-                "latitude":[42.25, 42.21],
-                "longitude":[-99.83, -99.32],
-                "time":'2014-09-06'},
-            dims=["longitude", "latitude"],
-            name='Temperature',
-            attrs=dict(units="degC"),
-        )
+        data=[[1.0, 2.0], [3.0, 4.0]],
+        coords={
+            "latitude": [42.25, 42.21],
+            "longitude": [-99.83, -99.32],
+            "time": '2014-09-06'},
+        dims=["longitude", "latitude"],
+        name='Temperature',
+        attrs=dict(units="degC"),
+    )
 
     dataarray = model.get_value_as_xarray("Temperature")
 
     xr.testing.assert_equal(dataarray, expected)
+
+
+class TestCheckParameterSet:
+    def test_correct_version(self, setup_config):
+        ps = ParameterSet(
+            name='justatest',
+            directory='justatest',
+            config='justatest/config.ini',
+            target_model='mockedmodel',  # == lowered class name
+            supported_model_versions={'0.4.2'}
+        )
+        m = MockedModel(parameter_set=ps)
+        assert m.parameter_set == ps
+
+    def test_wrong_model(self, setup_config):
+        ps = ParameterSet(
+            name='justatest',
+            directory='justatest',
+            config='justatest/config.ini',
+            target_model='wrongmodel',
+            supported_model_versions={'0.4.2'}
+        )
+        with pytest.raises(ValueError) as excinf:
+            MockedModel(parameter_set=ps)
+
+        expected = 'Parameter set has wrong target model'
+        assert expected in str(excinf.value)
+
+    def test_any_version(self, caplog, setup_config):
+        ps = ParameterSet(
+            name='justatest',
+            directory='justatest',
+            config='justatest/config.ini',
+            target_model='mockedmodel',  # == lowered class name
+            supported_model_versions=set()
+        )
+        with caplog.at_level(logging.WARNING):
```

Comment:
caplog is a pytest fixture?

Reply:
Yep see https://docs.pytest.org/en/latest/reference/reference.html?highlight=caplog#std-fixture-caplog